File: backend/app/models/job.py
import enum
from datetime import datetime

from sqlalchemy import DateTime, Enum, Float, ForeignKey, Integer, JSON, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from .base import Base


class JobStatus(str, enum.Enum):
    PENDING = "PENDING"
    RUNNING = "RUNNING"
    SUCCESS = "SUCCESS"
    FAILED = "FAILED"
    CANCELLED = "CANCELLED"


class JobExecution(Base):
    __tablename__ = "job_executions"

    id: Mapped[int] = mapped_column(primary_key=True)
    job_name: Mapped[str] = mapped_column(String(100), index=True)
    game_id: Mapped[int | None] = mapped_column(
        ForeignKey("game_definitions.id"), nullable=True
    )
    status: Mapped[JobStatus] = mapped_column(Enum(JobStatus))
    started_at: Mapped[datetime] = mapped_column(DateTime, index=True)
    finished_at: Mapped[datetime | None] = mapped_column(DateTime, nullable=True)
    duration_seconds: Mapped[float | None] = mapped_column(Float, nullable=True)
    result_summary: Mapped[dict | None] = mapped_column(JSON, nullable=True)
    error_message: Mapped[str | None] = mapped_column(Text, nullable=True)
    triggered_by: Mapped[str] = mapped_column(String(50))

    def __repr__(self) -> str:
        return f"<JobExecution(job={self.job_name!r}, status={self.status})>"

File: frontend/src/types/job.ts
export type JobStatus =
  | "PENDING"
  | "RUNNING"
  | "SUCCESS"
  | "FAILED"
  | "CANCELLED";

export interface JobExecution {
  id: number;
  job_name: string;
  game_id: number | null;
  status: JobStatus;
  started_at: string;
  finished_at: string | null;
  duration_seconds: number | null;
  result_summary: Record<string, unknown> | null;
  error_message: string | null;
  triggered_by: string;
}

export interface SchedulerStatus {
  running_jobs: string[];
  running_count: number;
  last_runs: Record<
    string,
    {
      status: string;
      started_at: string;
      finished_at: string | null;
      duration_seconds: number | null;
    }
  >;
}
